src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -221,58 +219,80 @@ class JooqBurdenEstimateRepository(
-                DISEASE.ID,
-                BURDEN_ESTIMATE.YEAR,
-                BURDEN_ESTIMATE.AGE,
-                COUNTRY.ID,
-                COUNTRY.NAME,
-                BURDEN_OUTCOME.CODE,
-                BURDEN_ESTIMATE.VALUE
-        )
-                .fromJoinPath(BURDEN_ESTIMATE_SET, BURDEN_ESTIMATE)
-                .join(RESPONSIBILITY)
-                .on(BURDEN_ESTIMATE_SET.RESPONSIBILITY.eq(RESPONSIBILITY.ID))
-                .joinPath(RESPONSIBILITY, SCENARIO, SCENARIO_DESCRIPTION, DISEASE)
-                .join(COUNTRY)
-                .on(BURDEN_ESTIMATE.COUNTRY.eq(COUNTRY.NID))
-                .joinPath(BURDEN_ESTIMATE, BURDEN_OUTCOME)
-                .where(BURDEN_ESTIMATE_SET.ID.eq(burdenEstimateSetId))
-                .orderBy(
-                        DISEASE.ID,
-                        BURDEN_ESTIMATE.YEAR,
-                        BURDEN_ESTIMATE.AGE,
-                        COUNTRY.ID,
-                        COUNTRY.NAME,
-                        BURDEN_OUTCOME.CODE)
-                .fetchSequence()
-                .map { it.into(BurdenEstimateOutcome::class.java) }
+        val cohortSize = dsl.select(BURDEN_OUTCOME.ID)
+                .from(BURDEN_OUTCOME)
+                .where(BURDEN_OUTCOME.CODE.eq("cohort_size"))
+                .fetchAnyInto(Short::class.java)
+
+        return countryLookup.keys.asSequence().map { k ->
+            val cursor = dsl.fetchLazy(getBurdenEstimateQueryForCountry(burdenEstimateSetId, outcomes, cohortSize, k))
+            generateSequence {
+                cursor.fetchNext()?.map { record ->
+                    val country = countryLookup.getValue(record.get("country", Short::class.java))
+                    val outcomeList = outcomes.filter { it.second != "cohort_size" }
+                            .associateBy({ it.second }, { record.get(it.second, Float::class.java) })
+                    BurdenEstimate(
+                            disease,
+                            record.get("year", Short::class.java),
+                            record.get("age", Short::class.java),
+                            country.first,
+                            country.second,
+                            record.get("cohort_size", Float::class.java),
+                            outcomeList)
+                }
+            }
+        }.flatten()
+    }
 
+    private fun getBurdenEstimateQueryForCountry(setId: Int,
+                                                 outcomes: List<Pair<Short, String>>,
+                                                 cohortSize: Short,
+                                                 country: Short): String
+    {
+        val outcomeIdQuery = "select ${outcomes.map { it.first }.joinToString(" union select ")} union select $cohortSize order by 1"
+        val expectedOutcomeCodes = "${outcomes.joinToString(" real,") { it.second }} real, cohort_size real"
+        val compoundRowKey = "year || '':''|| age || '':'' || country"
```

Comment:
Do you still need country in your compound key now you're only fetching results for one country??

Reply:
Oh nice, no I don't think we do